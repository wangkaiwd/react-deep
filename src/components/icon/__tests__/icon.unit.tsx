import renderer from 'react-test-renderer';
import Icon from '@/components/icon/icon';
import * as React from 'react';
import { mount } from 'enzyme';
// 测试引入scss文件的组件：@see:https://github.com/facebook/jest/issues/3094#issuecomment-284867098
describe('icon 组件', () => {
  it('渲染成功', () => {
    const json = renderer.create(<Icon name="alipay"/>);
    expect(json).toMatchSnapshot();
  });
  it('支持点击事件', () => {
    const fn = jest.fn();
    const component = mount(<Icon name="alipay" onClick={fn}/>);
    component.find('svg').simulate('click');
    expect(fn).toBeCalled();
  });
});
