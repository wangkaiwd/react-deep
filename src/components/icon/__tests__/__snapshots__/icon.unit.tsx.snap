// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon 组件 xxx 1`] = `
<svg
  className="wui-icon"
>
  <use
    xlinkHref="#alipay"
  />
</svg>
`;
